spectral_cube/spectral_cube.py

```diff
@@ -1912,6 +1912,122 @@ def subcube_from_ds9region(self, ds9region, allow_empty=False):
         # that we can now crop out
         return masked_subcube.minimal_subcube(spatial_only=True)
 
+    def subcube_from_ds9region_new(self, ds9region, allow_empty=False):
+        """
+        Extract a masked subcube from a ds9 region
+        (only functions on celestial dimensions)
+
+        Parameters
+        ----------
+        ds9region: str
+            The region to extract
+        allow_empty: bool
+            If this is False, an exception will be raised if the region
+            contains no overlap with the cube
+        """
+        import regions
+
+        if isinstance(ds9region, six.string_types):
+            ds9region = regions.DS9Parser(ds9region).shapes.to_regions()
+        else:
+            raise TypeError("{0} should be a DS9 string".format(ds9region))
+
+        return self.subcube_from_regions(ds9region, allow_empty)
+
+    def subcube_from_crtf(self, region, allow_empty=False):
+        """
+        Extract a masked subcube from a CRTF region
+        (only functions on celestial dimensions)
+
+        Parameters
+        ----------
+        region: str
+            The CRTF region string to extract
+        allow_empty: bool
+            If this is False, an exception will be raised if the region
+            contains no overlap with the cube
+        """
+        import regions
+
+        if isinstance(region, six.string_types):
+            region = regions.CRTFParser(region).shapes.to_regions()
+        else:
+            raise TypeError("{0} should be a CRTF string".format(region))
+
+        return self.subcube_from_regions(region, allow_empty)
+
+    def subcube_from_regions(self, region_list, allow_empty=False):
+        """
+        Extract a masked subcube from a list of `regions.Region` object
+        (only functions on celestial dimensions)
+
+        Parameters
+        ----------
+        region_list: `regions.Region` list
+            The region to extract
+        allow_empty: bool
+            If this is False, an exception will be raised if the region
+            contains no overlap with the cube
+        """
+        import regions
+
+        # Every region should be a `regions.PixelRegion` object.
```

Comment:
They aren't guaranteed to be yet.  Maybe rephrase the comment: "Convert all regions to `PixelRegion`s"

Reply:
that is why the following code converts `SkyRegion` objects to `PixelRegion` objects.
your rephrase is clearer though.